tools/tests/com.liferay.ide.project.core.tests/src/com/liferay/ide/project/core/tests/ProjectCoreBase.java

```diff
@@ -67,24 +71,41 @@
  * @author Gregory Amerson
  * @author Terry Jia
  * @author Simon Jiang
+ * @author Li Lu
  */
+@SuppressWarnings( "restriction" )
 public class ProjectCoreBase extends ServerCoreBase
 {
 
     private static final String bundleId = "com.liferay.ide.project.core.tests";
 
-    public static void deleteAllWorkspaceProjects() throws Exception
+    public static void deleteAllWorkspaceProjects()
     {
         for( IProject project : CoreUtil.getAllProjects() )
         {
-            if ( project != null && project.isAccessible() && project.exists())
+            if( project != null && project.isAccessible() && project.exists() )
             {
-                project.delete( true, true, new NullProgressMonitor() );
+                try
```

Comment:
This should not call delete 3 times, one time should be enough if it doesn't work that means we have some other operation that is still ongoing that hasn't finished that we need to stop.